include/fmt/format.h

```diff
@@ -3990,13 +3990,23 @@ FMT_FORMAT_AS(long, detail::long_type);
 FMT_FORMAT_AS(unsigned long, detail::ulong_type);
 FMT_FORMAT_AS(Char*, const Char*);
 FMT_FORMAT_AS(std::nullptr_t, const void*);
-FMT_FORMAT_AS(detail::std_string_view<Char>, basic_string_view<Char>);
 FMT_FORMAT_AS(void*, const void*);
 
 template <typename Char, typename Traits, typename Allocator>
 class formatter<std::basic_string<Char, Traits, Allocator>, Char>
     : public formatter<basic_string_view<Char>, Char> {};
 
+// Fixes issue #4036
+template <typename Char>
+struct formatter<detail::std_string_view<Char>, Char>
+    : public formatter<basic_string_view<Char>, Char> {
+  auto format(const detail::std_string_view<Char>& value,
+              format_context& ctx) const -> decltype(ctx.out()) {
+    using base = formatter<basic_string_view<Char>, Char>;
+    return base::format(value, ctx);
+  }
+};
```

Comment:
Looks like I forgot to actually comment what the requested change was =). Let's apply the change to `FMT_FORMAT_AS` rather than just the `formatter` specialization for `string_view`.

Reply:
No problem, have updated review (haven't squashed yet so you can see diffs), with what I believe you're saying. Just check the types of the params and this is what you meant etc.